R/helpers.R

```diff
@@ -0,0 +1,459 @@
+#' Helper function which calculates p-value via chi-square or fisher
+#'
+#' @param df <`tbl_df`> Dataframe that has variable and treatment columns of interest
+#' @param var <`character(1)`> Name of variable column
+#' @param treatment <`character(1)`> Name of treatment column
+#'
+#' @examples
+#'
+#' p_chi_fisher(mtcars, var = "am", treatment = "vs")
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_chi_fisher <- function(df, var, treatment) {
+
+  chisq_wrapper <- function(var, df, treatment) {
+    table <- stats::chisq.test(as.factor(df[[var]]), as.factor(df[[treatment]]))
+    return(table$p.value)
+  }
+
+  fisher_wrapper <- function(var, df, treatment) {
+    table <- stats::fisher.test(as.factor(df[[var]]), as.factor(df[[treatment]]), simulate.p.value = TRUE)
+    return(table$p.value)
+  }
+
+  chisq_wrapper <- purrr::quietly(chisq_wrapper)
+  chisq <- chisq_wrapper(var, df, treatment)
+
+  if (length(chisq$warnings) == 0) {
+    return(chisq$result)
+  } else {
+    return(fisher_wrapper(var, df, treatment))
+  }
+
+}
+
+#' Helper function which calculates p-value via t-test
+#'
+#' @inheritParams p_chi_fisher
+#'
+#' @examples
+#'
+#' p_t_test(mtcars, var = "mpg", treatment = "vs")
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_t_test <- function(df, var, treatment) {
+
+  df %>%
+    pull(var) %>%
+    split(df %>% pull(treatment)) %>%
+    stats::setNames(c("x", "y")) %>%
+    do.call(stats::t.test, .) %>%
+    `[[`("p.value")
+
+}
+
+
+#' Helper function which calculates p-value via anova
+#'
+#' @inheritParams p_chi_fisher
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_anova <- function(df, var, treatment) {
+
+  form <- stats::as.formula(paste0(var, " ~ ", treatment))
+  summary(stats::aov(form, df))[[1]]$`Pr(>F)`[1]
```

Comment:
Could this be rewritten to be easier to read?

Reply:
I need to dig more into the aov output. There has to be an easier way to access the p-value

Reply:
This should do it:
```
stats::aov(form, df) %>% stats::anova() %>% pluck("p.value")
```

Reply:
nevermind, it doesn't

Reply:
This also works
```
lm(form, df) %>%
  anova() %>%
  pull(`Pr(>F)`) %>%
  pluck(1)
```

Reply:
I'd use that, the list output from anova is clearly not tidy so I don't think there's a cleaner way

Reply:
We should also consider dropping the `p_t_test` and `p_wilcox` functions since the anova and kruskal wallis tests should match those for a two treatment table